#! /bin/sh -e
## 32_sb1_probe_ide.dpatch by Peter Horton <pdh@colonel-panic.org>
##
## DP: Make sure we scan SWARM IDE interface for devices.
## DP: Upstream status: submitted to linux-mips list

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

Index: linux.git/drivers/ide/mips/swarm.c
===================================================================
--- linux.git.orig/drivers/ide/mips/swarm.c	2006-01-23 22:48:58.000000000 +0000
+++ linux.git/drivers/ide/mips/swarm.c	2006-01-24 08:32:45.000000000 +0000
@@ -127,6 +127,7 @@
 	memcpy(hwif->io_ports, hwif->hw.io_ports, sizeof(hwif->io_ports));
 	hwif->irq = hwif->hw.irq;
 
+	probe_hwif_init(hwif);
 	dev_set_drvdata(dev, hwif);
 
 	return 0;

